### PR TITLE
bt-146: Create models and entities for files and badges migrations from user_profie

### LINKED ISSUE #146: Create models and entities for migrations from bt-72
Create models and entities migrations from bt-72
- [x] create models
- [x] create entities

### PR DESCRIPTION
**Pull request by nika-sirak**
Created models and entities. I need role to user model. Working on it.

### CODE REVIEW

File: backend/src/bundles/bsa-badges/bsa-badge.entity.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import { type Entity } from '~/common/types/types.js';
+
+class BSABadgeEntity implements Entity {
+    private 'id': string | null;
```
Comment (nikita-remeslov) [inline backend/src/bundles/bsa-badges/bsa-badge.entity.ts:4]: id can be null?

File: backend/src/bundles/files/file.model.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import {
+    AbstractModel,
+    DatabaseTableName,
+} from '~/common/database/database.js';
+
+class FileModel extends AbstractModel {
+    public 'url': string;
+
+    public 'file_name': string;
```
Comment (MaksymGavatuk) [inline backend/src/bundles/files/file.model.ts]:
```suggestion
    public 'fileName': string;
```

**Review by MaksymGavatuk — CHANGES_REQUESTED** [review]

File: backend/src/bundles/bsa-badges/bsa-badge.entity.ts (hunk 1 above)
Comment (lavandosya) [reply backend/src/bundles/bsa-badges/bsa-badge.entity.ts:4]: Yes, this needed, for example, for initializeNew(), where returns object with id: null. This approach was already in project for user.entity.ts

File: backend/src/bundles/talent-badges/talent-badge.model.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    public 'isShown': boolean;
+
+    public 'badgeId': string;
+
+    public 'userDetailsId': string | null;
+
+    public static override get tableName(): string {
+        return DatabaseTableName.TALENT_BADGES;
+    }
```
Comment (lavandosya) [inline backend/src/bundles/talent-badges/talent-badge.model.ts:25]:
```
public static get relationMappings(): RelationMappings {
        return {
            badge: {
                relation: Model.BelongsToOneRelation,
                modelClass: BSABadgesModel,
                join: {
                    from: `${DatabaseTableName.TALENT_BADGES}.badgeId`,
                    to: `${DatabaseTableName.BSA_BADGES}.id`,
                },
            },
        };
    }
```

**Review by lavandosya — APPROVED** [review]

**Review by MaksymGavatuk — APPROVED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**